OpenQASM introduction page (#295)
Qiskit/documentation: docs/build/introduction-to-qasm.mdx

Reviewer: The only thing I feel is missing from this piece is some code. Would it be possible to include a section that just shows an example of what a typical qasm program looks like and how to interpret the syntax? Doesn't have to be a lot, just something to give readers a flavour and then they can dig into the paper in more detail

---
title: Introduction to OpenQASM
description: An introduction to OpenQASM (Open quantum assembly language)
---

# Introduction to OpenQASM

> OpenQASM (open quantum assembly language), an imperative programming language for describing quantum circuits, is a platform-agnostic (or is "machine-independent" better?) programming interface for IBM Quantum systems. OpenQASM uses the measurement-based quantum circuit model to express quantum programs as ordered sequences of parameterized operations (such as gates, measurements, and resets) and real-time classical computation. OpenQASM can also describe circuits intended to characterize, validate, or debug quantum systems.

Author: ```suggestion
OpenQASM (open quantum assembly language),  a machine-independent programming interface for IBM Quantum systems, is an imperative programming language for describing quantum circuits. OpenQASM uses the quantum circuit model to express quantum programs as ordered sequences of parameterized operations (such as gates, measurements, and resets) and real-time classical computation. In addition to quantum algorithms, OpenQASM can describe circuits intended to characterize, validate, or debug quantum systems.
```

---
title: Introduction to OpenQASM
description: An introduction to OpenQASM (Open quantum assembly language)
---

# Introduction to OpenQASM

> OpenQASM (open quantum assembly language), a machine-independent programming interface for IBM Quantum systems, is an imperative programming language for describing quantum circuits. OpenQASM uses the quantum circuit model to express quantum programs as ordered sequences of parameterized operations (such as gates, measurements, and resets) and real-time classical computation. In addition to quantum algorithms, OpenQASM can describe circuits intended to characterize, validate, or debug quantum systems.

Reviewer 2: This turn of phrase sounds like OpenQASM is only for IBM systems: "a machine-independent programming interface for IBM Quantum systems". Or at the very least, it sounds like an oxymoron. Rephrase?

Author: How about "a machine-independent programming interface compatible with IBM Quantum Systems"?

Reviewer 2: Yes, that works.

> OpenQASM (open quantum assembly language), a machine-independent programming interface for IBM Quantum systems, is an imperative programming language for describing quantum circuits. OpenQASM uses the quantum circuit model to express quantum programs as ordered sequences of parameterized operations (such as gates, measurements, and resets) and real-time classical computation. In addition to quantum algorithms, OpenQASM can describe circuits intended to characterize, validate, or debug quantum systems.

Author: ```suggestion
OpenQASM (open quantum assembly language), a machine-independent programming interface compatible with IBM Quantum systems, is an imperative programming language for describing quantum circuits. OpenQASM uses the quantum circuit model to express quantum programs as ordered sequences of parameterized operations (such as gates, measurements, and resets) and real-time classical computation. In addition to quantum algorithms, OpenQASM can describe circuits intended to characterize, validate, or debug quantum systems.
```
